src/ptests/test_unstruc_quad.c

```diff
@@ -185,9 +207,46 @@ int main(int argc, char* argv[]) {
   if (cgp_elements_write_data(fn,B,Z,S,emin,emax,el_ptr))
     cgp_error_exit();
 
+  n_boco_elems = 1;
+  if (cg_boco_write(fn, B, Z, "Left BC", CGNS_ENUMV(BCTypeUserDefined), CGNS_ENUMV(PointList), n_boco_elems, NULL, &BC))
+    cgp_error_exit();
+
+  if (comm_rank == 0) {
+    start_local = 1;
+    end_local   = 1;
+
+    elements[0] = start;
+    el_ptr      = elements;
+  } else {
+    start_local = 0;
+    end_local   = 0;
+
+    el_ptr = NULL;
+  }
+
+  printf("%d: %d %d\n", comm_rank, (int)start_local, (int)end_local);
+  cg_goto(fn, B, "Zone_t", Z, "ZoneBC_t", 1, "BC_t", BC, "PointList", 0, NULL);
+  if (cgp_ptlist_write_data(fn, start_local, end_local, el_ptr))
+    cgp_error_exit();
+  if (cg_boco_gridlocation_write(fn, B, Z, BC, CGNS_ENUMV(EdgeCenter)))
+    cgp_error_exit();
+
+  if (cgp_close(fn)) cgp_error_exit();
+
+  // Test file reading
+  if (cgp_open("test_unstruc_quad.cgns", CG_MODE_READ, &fn)) cgp_error_exit();
+
+  for (int i = 0; i < nelem*4; i++) elements[i] = 0;
+  start_local = comm_rank * 3 + 1;
+  end_local = start_local + 2;
+  if (cg_goto(fn, B, "Zone_t", Z, "ZoneBC_t", 1, "BC_t", 1, "PointList", 0, NULL)) cgp_error_exit();
+  if (cgp_ptlist_read_data(fn, start_local, end_local, elements)) cgp_error_exit();
+  printf("%d: %d, %d, %d\n", comm_rank, (int)elements[0], (int)elements[1], (int)elements[2]);
+
```

Comment:
It would be good to check the read values are correct instead of printing them. I'm working on changing the tests from printing to value checking.

Reply:
Done